sdk/azidentity/client_certificate_credential_test.go

```diff
@@ -165,14 +157,8 @@ func TestClientCertificateCredential_NoCertificate(t *testing.T) {
 
 func TestClientCertificateCredential_NoPrivateKey(t *testing.T) {
 	test := allCertTests[0]
-	srv, close := mock.NewTLSServer()
-	defer close()
-	srv.AppendResponse(mock.WithBody(accessTokenRespSuccess))
-	options := ClientCertificateCredentialOptions{}
-	options.Cloud.ActiveDirectoryAuthorityHost = srv.URL()
-	options.Transport = srv
 	var key crypto.PrivateKey
-	_, err := NewClientCertificateCredential(fakeTenantID, fakeClientID, test.certs, key, &options)
+	_, err := NewClientCertificateCredential(fakeTenantID, fakeClientID, test.certs, key, nil)
```

Comment:
How come this one doesn't need a `&mockSTS{}` in the options?

Reply:
Oh perhaps because the inputs cause the constructor to fail.

Reply:
The constructor doesn't do any I/O and should return an error when the private key is nil